rslib/src/error/not_found.rs

```diff
@@ -0,0 +1,44 @@
+// Copyright: Ankitects Pty Ltd and contributors
+// License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
+
+use std::{any, fmt};
+
+use snafu::{Backtrace, OptionExt, Snafu};
+
+use crate::prelude::*;
+
+#[derive(Debug, Snafu)]
+#[snafu(visibility(pub), display("{message}"))]
+pub struct NotFoundError {
+    pub message: String,
+    pub backtrace: Option<Backtrace>,
+}
+
+impl PartialEq for NotFoundError {
+    fn eq(&self, other: &Self) -> bool {
+        self.message == other.message
+    }
+}
+
+impl Eq for NotFoundError {}
+
+/// Allows generating [AnkiError::NotFound] from [Option::None] and the
+/// typical [core::result::Result::Err].
+pub trait OkOrNotFound {
+    type Value;
+    fn ok_or_not_found(self, identifier: impl fmt::Debug) -> Result<Self::Value>;
+}
+
+impl<T> OkOrNotFound for Option<T> {
+    type Value = T;
+
+    fn ok_or_not_found(self, identifier: impl fmt::Debug) -> Result<Self::Value> {
+        self.with_context(|| NotFoundSnafu {
+            message: format!(
+                "no such {}: {identifier:?}",
+                any::type_name::<Self::Value>()
```

Comment:
A minor nicety would be to split on `::` and only use the trailing portion, and maybe lowercase too.

Reply:
I guess for case conversion we should do it properly with [convert_case](https://docs.rs/convert_case/latest/convert_case/#).

Reply:
Could we use the existing inflections crate?

Reply:
Just to make sure: inflections is only a dependency of the i18n crate currently, whereas convert_case is already a sub dependency of the main crate. Do you still prefer adding the former?

Reply:
I'm not seeing convert_case in the output of 'cargo tree' - what's leading you to think we're already depending on it?

Reply:
Sorry, I see it's also not a sub dependency of the anki crate, but of linkchecker.

Reply:
Ah, gotcha. Thanks!